airflow/models/dag.py

```diff
@@ -1375,6 +1375,11 @@ def get_is_paused(self, session=NEW_SESSION) -> None:
         """Return a boolean indicating whether this DAG is paused."""
         return session.scalar(select(DagModel.is_paused).where(DagModel.dag_id == self.dag_id))
 
+    @provide_session
+    def get_last_parsed_time(self, session=NEW_SESSION) -> None:
```

Comment:
This does not return None (well it can but not always) so the return type annotation needs to be fixed. Otherwise looks god to me.

Reply:
I pushed a fix to this typing!